mk/opencl.mk

```diff
@@ -1,7 +1,13 @@
-# FIXME: perform sanity checks before compilation
-OPENCL_LIB ?= /usr/local/cuda-9.1/lib64
+OPENCL_LIB_PATH ?= /usr/local/cuda/lib64
+
+OPENCL_LIB := $(shell test -d $(OPENCL_LIB_PATH); echo $$?)
+
+ifneq ($(OPENCL_LIB),0)
+    $(error "Please specify the path to your OpenCL library on your platform")
+endif
 
 CFLAGS += \
         -DENABLE_OPENCL \
-        -I$(OPENCL_PATH)/include
-LDFLAGS += -L$(OPENCL_LIB) -lOpenCL
+        -I$(OPENCL_LIB_PATH)/../include
+
+LDFLAGS += -L$(OPENCL_LIB_PATH) -lOpenCL
```

Comment:
Before assigning `$(LDFLAGS)`, we shall ensure the presence of OpenCL runtime library in advance.

Reply:
Do you mean check the presence of `libOpenCL.so` ?

Reply:
Yes, exactly.

Reply:
done.